py_cubic_ods_ingestion/tests/test_job_helpers.py

```diff
@@ -4,22 +4,139 @@
 
 import json
 import pytest
+from pyspark.sql import SparkSession
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession as SparkSessionType
 
 from py_cubic_ods_ingestion import job_helpers
 
+# helper functions
+def collection_as_dict(collection: list) -> map:
+    """
+    Convert from list of Spark Rows to list of dicts.
 
-def test_removeprefix() -> None:
+    Parameters
+    ----------
+    collection : list
+        List of Spark Rows
+
+    Returns
+    -------
+    map
+        An iterable list of dicts
     """
-    Testing removing of prefix from string.
+
+    return map(lambda item: item.asDict(), collection)
```

Comment:
question: the return type doesn't look right here: `map` is a function? If you way to say it returns an iterable of dicts, that's `Iterable[Dict[Any, Any]]` (or a more specific dict type if you have it).

suggestion:
```suggestion
    return [item.asDict() for item in collection]
```
or (lazy version)
```suggestion
    for item in collection:
        yield item.asDict()
```

Reply:
Fixed with your first suggestion.